pkg/bindings/connection.go

```diff
@@ -154,8 +155,20 @@ func pingNewConnection(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
 	if response.StatusCode == http.StatusOK {
-		return nil
+		v, err := semver.ParseTolerant(response.Header.Get("Libpod-API-Version"))
+		if err != nil {
+			return err
+		}
+
+		switch APIVersion.Compare(v) {
+		case 1, 0:
+			// Server's job when client version is equal or older
+			return nil
+		case -1:
+			return errors.Errorf("server API version is too old. client %s server %s", APIVersion.String(), v.String())
```

Comment:
I enjoy quoted strings in logs:
```suggestion
			return errors.Errorf("server API version is too old. client %q server %q", APIVersion.String(), v.String())
```